jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -258,14 +276,17 @@ private boolean select()
             }
             catch (Throwable x)
             {
+                Selector selector = _selector;
                 if (isRunning())
-                    LOG.warn(x);
+                {
+                    onSelectFailed(x);
+                }
                 else
                 {
                     LOG.warn(x.toString());
                     LOG.debug(x);
                 }
-                closeNoExceptions(_selector);
+                closeNoExceptions(selector);
```

Comment:
Is this good enough to give close/fail notifications to all the registered EndPoints? 
I don't think it is - so we need to either ensure the endpoints are properly closed or somehow pass them to the new selector.

Reply:
A SelectorProducer.select() failure has never notified registered endpoints, that would be new functionality.

The method `closeNoExceptions(Closeable)` is identical to `IO.close(Closeable)` (and in fact, could be removed from ManagedSelector)

I believe the notification of registered endpoints occurs via one of the other tasks, like DestroyEndPoint, which I don't think would work if the selector thread is dead.

Reply:
We never notified endpoints on select failure before because we always thought that could not happen in anyway other than a fatal exception.    But if are treating it as non fatal, then we have to either rescue the existing endpoints or close them properly. We can't just continue with hanging endpoints.

Reply:
Since we would need to ask the selector what the endpoints are, we cannot reliably get the endpoints on a dead selector. 
The `Selector.keys()` will throw an exception.